specs/DataSet.rst

```diff
@@ -0,0 +1,253 @@
+=====================
+DataSet Specification
+=====================
+
+Introduction
+============
+
+The DataSet class is used in QCoDeS to hold measurement results.
+It is the destination for measurement loops and the source for plotting and data analysis.
+As such, it is a central component of QCoDeS.
+
+The DataSet class should be usable on its own, without other QCoDeS components.
+In particular, the DataSet class should not require the use of Loop and parameters, although it should integrate with those components seamlessly.
+This will significantly improve the modularity of QCoDeS by allowing users to plug into and extend the package in many different ways.
+As long as a DataSet is used for data storage, users can freely select the QCoDeS components they want to use.
+
+Terminology
+================
+
+Parameter
+    A logically-single value input to or produced by a measurement.
+    A parameter need not be a scalar, but can be an array or a tuple or an array of tuples, etc.
+    A DataSet parameter corresponds conceptually to a QCoDeS parameter, but does not have to be defined by or associated with a QCoDeS Parameter . 
+    Roughly, a parameter represents a column in a table of experimental data.
+    
+Result
+    A result is the collection of parameter values associated to a single measurement in an experiment.
+    Roughly, a result corresponds to a row in a table of experimental data.
+    
+Role
+    Parameters may play different roles in a measurement.
+    Specifically, they may be input to the measurement (set-points) or outputs of the measurement (measured or computed values).
+    This distinction is important for plotting and for replicating an experiment.
```

Comment:
I get the impression that some users would strongly prefer that there is a default plot representation of 1D and 2D datasets. However this could probably just as well be handled by writing some metadata to the dataset naming the default x and y axis in the dataset (naturally over-writable for more advanced plots)